app/models/organisation.rb

```diff
@@ -0,0 +1,15 @@
+# == Schema Information
+#
+# Table name: organisation
+#
+#  id     :integer          not null, primary key
+#  name   :text
+#  org_id :text
+#
+
+class Organisation < ApplicationRecord
+  has_many :organisation_users
+  has_many :users, through: :organisation_users
+  has_many :organisation_providers
+  has_many :providers, through: :organisation_providers
```

Comment:
Probably dumb question, but can't this be stripped to:

```ruby
	  has_many :users, through: :organisation_users
	  has_many :providers, through: :organisation_providers
```

Do we need the has_many: join model ones?

Reply:
I think for the `has_many ... through` syntax you need an existing association, but here I think what we really want is a [`has_and_belongs_to_many` association](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_and_belongs_to_many). So far as I can tell `organisation_user` and `organisation_provider` are simple join-tables.

Reply:
Yeah, I was thinking there is a terser way to do it, HABTM is probably what we want :)